### PR TITLE
OEL-760: [BCL upgrade] - Date block composition

### PR DESCRIPTION
**Pull request by abel-santos-corral**
Jira issue(s):
- [OEL-760](https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-760)

### CODE REVIEW

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+date_block:
+  label: "Date Block"
+  description: "Date block allows to present a date block that can be placed in other components."
+  variants:
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: Not needed at this moment. A single variant if not a variant is default behaviour.

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    default:
+      label: Default
+  fields:
+    day:
+      type: "number"
+      label: "Day"
+      preview: "27"
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: We can add descriptions for all fields, so in the patterns example page is all more clear and is useful.

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    month:
+      type: "number"
+      label: "Month"
+      preview: "MAY"
+    year:
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: Add description

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  variants:
+    default:
+      label: Default
+  fields:
+    day:
+      type: "number"
+      label: "Day"
+      preview: "27"
+    month:
+      type: "number"
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: Add description

File: templates/patterns/date_block/pattern-date-block.html.twig [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{#
+/**
+ * @file
+ * Date-block.
+ */
+#}
+{% set attributes = create_attribute() %}
+
+{% include '@oe-bcl/bcl-date-block/date-block.html.twig' with {
+  'variant': variant,
```
Comment (escuriola) [inline templates/patterns/date_block/pattern-date-block.html.twig]: Variant is not needed

File: tests/src/Kernel/fixtures/markup_rendering_patterns/date_block.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    count:
+      'time.bcl-date-block.d-flex.flex-column.align-items-center.justify-content-center.bg-date.text-light.pt-3.rounded-top.mw-date': 1
+      'time[datetime="2021-05-27"]': 1
+      'span[class="pb-3 text-uppercase"]': 1
+      'span[class="bg-light w-100 text-center py-2 text-dark rounded-bottom mb-n1"]': 1
+    equals:
+      'span.pb-3.text-uppercase': "MAY"
+      'span[class="bg-light w-100 text-center py-2 text-dark rounded-bottom mb-n1"]': "2021"
+    contains:
+      'span__day': "27"
```
Comment (escuriola) [inline tests/src/Kernel/fixtures/markup_rendering_patterns/date_block.yml:20]: Add more tests with another date format.

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      label: "Month"
+      preview: "MAY"
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: Month is type "number" but the preview is "MAY"?

File: templates/patterns/date_block/date_block.ui_patterns.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    year:
+      type: "text"
```
Comment (escuriola) [inline templates/patterns/date_block/date_block.ui_patterns.yml]: Year is type "text" when month and day are type "number"?

File: templates/patterns/date_block/date_block.ui_patterns.yml (hunk 2 above)
Comment (abel-santos-corral) [reply templates/patterns/date_block/date_block.ui_patterns.yml]: I removed, that doesn't give more info, is the day number...

**Review by escuriola — DISMISSED** [review]

File: templates/patterns/date_block/pattern-date-block.html.twig [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{#
+/**
+ * @file
+ * Date-block.
```
Comment (drishu) [inline templates/patterns/date_block/pattern-date-block.html.twig]:
```suggestion
 * Date block pattern.
```

File: templates/patterns/date_block/pattern-date-block.html.twig [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+ */
+#}
+{% set attributes = create_attribute() %}
+
+{% include '@oe-bcl/bcl-date-block/date-block.html.twig' with {
+  'day': day,
+  'month': month,
+  'date_time': date_time,
+  'year': year,
+  'attributes': attributes,
```
Comment (drishu) [inline templates/patterns/date_block/pattern-date-block.html.twig]: what's the point if it's an empty object ?

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/patterns/date_block/pattern-date-block.html.twig (hunk 10 above)
Comment (abel-santos-corral) [reply templates/patterns/date_block/pattern-date-block.html.twig:14]: I'll have a call to explain that. In bcl component date_block there's no initializing of attributes object, so when adding the classes these got lost. So we moved here the initialize of the object so we can pass it to the component and it gets painted correctly.

**Review by drishu — APPROVED** [review]

**Review by drishu — DISMISSED** [review]

**Pull request closed without merging**